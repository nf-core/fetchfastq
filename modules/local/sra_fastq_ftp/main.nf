
process SRA_FASTQ_FTP {
    tag "$meta.id"
    label 'process_low'
    label 'error_retry'

    conda "conda-forge::wget=1.20.1"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/wget:1.20.1' :
        'biocontainers/wget:1.20.1' }"

    input:
    meta    : Map<String,String>

    output:
    fastq_1 : Path  = file('*_1.fastq.gz')
    fastq_2 : Path? = file('*_2.fastq.gz')
    md5_1   : Path  = file('*_1.fastq.gz.md5')
    md5_2   : Path? = file('*_2.fastq.gz.md5')

    topic:
    ( task.process, 'wget', eval("echo \$(wget --version | head -n 1 | sed 's/^GNU Wget //; s/ .*\$//')") ) >> 'versions'

    script:
    def args = task.ext.args ?: ''
    if (meta.single_end.toBoolean()) {
        """
        wget \\
            $args \\
            -O ${meta.id}.fastq.gz \\
            ${meta.fastq_1}

        echo "${meta.md5_1}  ${meta.id}.fastq.gz" > ${meta.id}.fastq.gz.md5
        md5sum -c ${meta.id}.fastq.gz.md5
        """
    } else {
        """
        wget \\
            $args \\
            -O ${meta.id}_1.fastq.gz \\
            ${meta.fastq_1}

        echo "${meta.md5_1}  ${meta.id}_1.fastq.gz" > ${meta.id}_1.fastq.gz.md5
        md5sum -c ${meta.id}_1.fastq.gz.md5

        wget \\
            $args \\
            -O ${meta.id}_2.fastq.gz \\
            ${meta.fastq_2}

        echo "${meta.md5_2}  ${meta.id}_2.fastq.gz" > ${meta.id}_2.fastq.gz.md5
        md5sum -c ${meta.id}_2.fastq.gz.md5
        """
    }
}
