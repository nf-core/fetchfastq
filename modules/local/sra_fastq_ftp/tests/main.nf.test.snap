{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "single_end": false,
                            "md5_1": "89c5be920021a035084d8aeb74f32df7",
                            "md5_2": "56271be38a80db78ef3bdfc5d9909b98"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz:md5,baaaea61cba4294ec696fdfea1610848",
                            "SRX9626017_SRR13191702_2.fastq.gz:md5,8e43ad99049fabb6526a4b846da01c32"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "single_end": false,
                            "md5_1": "89c5be920021a035084d8aeb74f32df7",
                            "md5_2": "56271be38a80db78ef3bdfc5d9909b98"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz.md5:md5,055a6916ec9ee478e453d50651f87997",
                            "SRX9626017_SRR13191702_2.fastq.gz.md5:md5,c30ac785f8d80ec563fabf604d8bf945"
                        ]
                    ]
                ],
                "2": [
                    "versions.yml:md5,7bfc86ca1f3e3236dbb91eb85c1d7af0"
                ],
                "fastq": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "single_end": false,
                            "md5_1": "89c5be920021a035084d8aeb74f32df7",
                            "md5_2": "56271be38a80db78ef3bdfc5d9909b98"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz:md5,baaaea61cba4294ec696fdfea1610848",
                            "SRX9626017_SRR13191702_2.fastq.gz:md5,8e43ad99049fabb6526a4b846da01c32"
                        ]
                    ]
                ],
                "md5": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "single_end": false,
                            "md5_1": "89c5be920021a035084d8aeb74f32df7",
                            "md5_2": "56271be38a80db78ef3bdfc5d9909b98"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz.md5:md5,055a6916ec9ee478e453d50651f87997",
                            "SRX9626017_SRR13191702_2.fastq.gz.md5:md5,c30ac785f8d80ec563fabf604d8bf945"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7bfc86ca1f3e3236dbb91eb85c1d7af0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-17T09:16:57.715689"
    }
}